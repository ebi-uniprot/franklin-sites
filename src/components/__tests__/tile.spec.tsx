/* eslint-disable jsx-a11y/anchor-has-content */
/* eslint-disable jsx-a11y/control-has-associated-label */
import { render } from '@testing-library/react';

import Tile from '../tile';
import { ProtVistaIcon } from '..';

import colors from '../../styles/colours.json';

describe('Tile component', () => {
  it('should render', () => {
    const { asFragment } = render(
      <Tile
        title="Tile title"
        subtitle="Subtitle"
        backgroundImage={<ProtVistaIcon />}
        backgroundColor={colors.seaBlue}
        link={<a href="https://www.uniprot.org" />}
        gradient
      >
        My description
      </Tile>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with default specified width', () => {
    const { asFragment } = render(
      <Tile
        title="Tile title"
        width="20rem"
        link={<a href="https://www.uniprot.org" />}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
