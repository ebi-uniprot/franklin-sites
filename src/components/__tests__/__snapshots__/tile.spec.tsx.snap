// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile component should render 1`] = `
<DocumentFragment>
  <div
    class="tile tile-gradient tile--has-link"
    style="--tile-background: #00639a;"
  >
    <a
      aria-hidden="true"
      href="https://www.uniprot.org"
      tabindex="-1"
    />
    <div
      aria-hidden="true"
      class="tile__background-image"
    >
      <test-file-stub />
    </div>
    <div
      class="tile__main-content"
    >
      <h2
        class="tile__header big"
      >
        Tile title
      </h2>
      <h3
        class="tile__subtitle small"
      >
        Subtitle
      </h3>
    </div>
    <small
      class="tile__description"
    >
      My description
    </small>
  </div>
</DocumentFragment>
`;

exports[`Tile component should render with default specified width 1`] = `
<DocumentFragment>
  <div
    class="tile tile--has-link"
    style="width: 20rem;"
  >
    <a
      aria-hidden="true"
      href="https://www.uniprot.org"
      tabindex="-1"
    />
    <div
      aria-hidden="true"
      class="tile__background-image"
    />
    <div
      class="tile__main-content"
    >
      <h2
        class="tile__header big"
      >
        Tile title
      </h2>
    </div>
  </div>
</DocumentFragment>
`;
